packages/block-editor/src/components/inserter/tabs.native.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import { Animated, Dimensions, View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useCallback,
+	useEffect,
+	useMemo,
+	useRef,
+	useState,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { SegmentedControl } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import BlocksTypesTab from './blocks-types-tab';
+import ReusableBlocksTab from './reusable-blocks-tab';
+import styles from './style.scss';
+
+const TAB_ANIMATION_DURATION = 250;
+
+function InserterTabs( {
+	listProps,
+	onSelect,
+	rootClientId,
+	showReusableBlocks,
+	tabIndex,
+} ) {
+	const [ windowWidth, setWindowWidth ] = useState(
+		Dimensions.get( 'window' ).width
+	);
+	const tabAnimation = useRef( new Animated.Value( 0 ) ).current;
+	const lastScrollEvents = useRef( [] ).current;
+
+	useEffect( () => {
+		Dimensions.addEventListener( 'change', onDimensionsChange );
+		return () =>
+			Dimensions.removeEventListener( 'change', onDimensionsChange );
+	}, [] );
+
+	useEffect( () => {
+		Animated.timing( tabAnimation, {
+			duration: TAB_ANIMATION_DURATION,
+			toValue: tabIndex,
+			useNativeDriver: true,
+		} ).start();
+
+		// Notify upstream with the last scroll event of the current tab.
+		const lastScrollEvent = lastScrollEvents[ tabIndex ];
+		if ( lastScrollEvent ) {
+			listProps.onScroll( { nativeEvent: lastScrollEvent } );
+		}
+	}, [ tabIndex ] );
+
+	const { tabs, tabKeys } = useMemo( () => {
+		const filteredTabs = InserterTabs.TABS.filter(
+			( { name } ) => showReusableBlocks || name !== 'reusable'
+		);
+		return {
+			tabs: filteredTabs,
+			tabKeys: [ ...filteredTabs.keys() ],
+		};
+	}, [ showReusableBlocks ] );
+
+	const translateX = useMemo(
+		() =>
+			tabKeys.length > 1
+				? tabAnimation.interpolate( {
+						inputRange: tabKeys,
+						outputRange: tabKeys.map(
+							( key ) => key * -windowWidth
+						),
+				  } )
+				: tabAnimation,
+		[ tabAnimation, tabKeys, windowWidth ]
+	);
+
+	function onScroll( event ) {
+		lastScrollEvents[ tabIndex ] = event.nativeEvent;
+		listProps.onScroll( event );
+	}
+
+	function onDimensionsChange() {
+		setWindowWidth( Dimensions.get( 'window' ).width );
+	}
+
+	return (
+		<View style={ styles[ 'inserter-tabs__wrapper' ] }>
+			<Animated.View
+				style={ [
+					styles[ 'inserter-tabs__container' ],
+					{
+						width: windowWidth * tabKeys.length,
+						transform: [ { translateX } ],
+					},
+				] }
```

Comment:
Could we move this style to a const?

Reply:
Sure! I'm thinking that I might even memoize it to prevent potential re-renders, wdyt?

Reply:
I moved these styles to a constant in this [commit](https://github.com/WordPress/gutenberg/pull/28495/commits/6153e92ad76c42ad40d89101b44ff0cafa731fc5), I finally decided not to memoize it because we don't have too many re-renders here, anyways if you think it would be worth it, I could apply it.

Reply:
I think it's ok to not memoize them 👍 Thanks for the change!